### DISCUSSION TITLE
How to add custom_metric

### DISCUSSION THREAD

**Discussion by hanhanwu** [microsoft/FLAML#477] [Q&A]
Hello FLAML team,

I'm trying to do HPO with customized metrics.
My customized metric does have the same signature ask this documentation requested: https://microsoft.github.io/FLAML/docs/reference/automl/

Here's my code:
![Screen Shot 2022-03-10 at 11 54 42 AM](https://user-images.githubusercontent.com/4024769/157744866-29a05c9c-2827-4edb-a529-dcb2cbc7df9d.png)

But tried multiple settings, not sure where to add this customized metric in automl... Got this type of error:
![Screen Shot 2022-03-10 at 11 55 45 AM](https://user-images.githubusercontent.com/4024769/157745046-1b68f933-f1ed-43bc-825a-d9133ccccfd9.png)

Do you know how to solve this problem? It's kinda urgent for me... How to hear from you soon... Thank you!

**Comment by sonichi** [comment]
> Hello FLAML team,
> 
> I'm trying to do HPO with customized metrics. My customized metric does have the same signature ask this documentation requested: https://microsoft.github.io/FLAML/docs/reference/automl/
> 
> Here's my code: ![Screen Shot 2022-03-10 at 11 54 42 AM](https://user-images.githubusercontent.com/4024769/157744866-29a05c9c-2827-4edb-a529-dcb2cbc7df9d.png)
> 
> But tried multiple settings, not sure where to add this customized metric in automl... Got this type of error: ![Screen Shot 2022-03-10 at 11 55 45 AM](https://user-images.githubusercontent.com/4024769/157745046-1b68f933-f1ed-43bc-825a-d9133ccccfd9.png)
> 
> Do you know how to solve this problem? It's kinda urgent for me... How to hear from you soon... Thank you!

Please change the signature into:
```
def custom_metric(
    X_val, y_val, estimator, labels,
    X_train, y_train, weight_val=None, weight_train=None,
    *args,
):
```
We'll update the documentation.

**Reply by hanhanwu** [reply]
Thank you @sonichi ! Now the code can finish HPO work. But when I'm trying to check the output, got this error:
![Screen Shot 2022-03-10 at 2 51 05 PM](https://user-images.githubusercontent.com/4024769/157768517-e696726f-b1b0-4262-ade5-4e048c0c6a13.png)

Looks like the result is the same as an empty AutoML instance... Do you know how to resolve this? Thank you!

**Reply by sonichi** [reply]
Has at least one trial finished within the budget? Could you check the output of the AutoML run?

**Reply by hanhanwu** [reply]
Ohh! It's because my divisor in the customized metric was 0.... Now I got the output. Thank you very much @sonichi !
